### DISCUSSION TITLE
v2.2.5 message "Invalid cross duration: 0.00 <= 0.!"

### DISCUSSION THREAD

**Discussion by ArvyRogerio** [savonet/liquidsoap#4007] [General]
Hi, I'm having problems using "annotate" in version 2.2.5. Previous version that I was using (2.1.4) I had no problems.

Error **[cross:1] Invalid cross duration: 0.00 <= 0.!**

Request: _annotate:liq_cue_in=0.,liq_cross_duration=1.67:/path/11451.mp3_

And I feel a little "skip" in the very beginning of the song, a small "jump". If I don't use "annotate", songs play perfect.

The error still appears if I change liq_cue_in to 0.01 for example.

Any ideas?

**Comment by vitoyucepi** [comment]
Hi @ArvyRogerio,
I can't say for sure, but it seems that something changed between 2.2.x and 2.3.x, so it would probably be fixed in the next release.
Consider creating a complete issue anyway.

**Comment by Moonbase59** [comment]
> Request: _annotate:liq_cue_in=0.,liq_cross_duration=1.67:/path/11451.mp3_

@ArvyRogerio Have you tried using `liq_cross_start_next` (float seconds from start of file) instead of `liq_cross_duration`?

Nevertheless, I’ve also heard about problems with `liq_cue_in` being very short and typically use a value of `0.1` myself (or whatever Autocue calculates).

**Reply by ArvyRogerio** [reply]
I'll try soon, thanks!

**Reply by ArvyRogerio** [reply]
@Moonbase59  No difference using `liq_cross_start_next` ...

**Comment by toots** [comment]
Hi! Do you have more extracts from the logs?

**Reply by ArvyRogerio** [reply]
The "skip" that I mentioned is not confirmed, I need to test more.

> 2024/07/08 12:10:18 [decoder.mad:3] Libmad recognizes "/iwradio/arquivos/musicas/2407/11453.mp3" as mpeg audio (layer III, 192kbps, 48000Hz, 2 channels).
> 2024/07/08 12:10:18 [lang:2] ---- Adding: annotate:liq_cross_start_next=0.1,liq_cross_duration=0.35:/iwradio/arquivos/musicas/2407/11453.mp3
> 2024/07/08 12:10:27 [decoder:2] Decoding "/iwradio/arquivos/musicas/2407/11451.mp3" ended: Mad.End_of_stream.
> 2024/07/08 12:10:28 [cross:1] Invalid cross duration: 0.00 <= 0.!
> 2024/07/08 12:10:28 [musicas:3] Prepared "/iwradio/arquivos/musicas/2407/11453.mp3" (RID 7).
> 2024/07/08 12:10:28 [cross:3] Analysis: -30.491417dB / -17.900335dB (1.66s / 1.66s)
> 2024/07/08 12:10:28 [cross:3] Computing crossfade transition over overlapping 1.66s buffered data at start and end.
> 2024/07/08 12:10:28 [cross_musicas:3] Simple transition: crossed, fade-in, fade-out.

**Comment by ArvyRogerio** [comment]
Well, I returned to the format `annotate:liq_cue_in=0.,liq_cross_duration=1.67:/path/11451.mp3`.

Now I realized something: the songs are not "mixing" smoothly (overlap). When `liq_cross_duration` triggers, the "old" song is cutted, and the new is inserted. So, my question is, something related to this changed since version 2.1.3?

So, for version 2.2.5, what are the correct combination of liq_* parameters to smoothly mix the songs? Or need any operator, or change?

@Moonbase59 Any ideas? Thanks.

-----------

I have some "stations IDs" and now I have  a new message:

> **2024/07/08 17:01:42 [lang:2] ---- ID: annotate:liq_cue_in=0.12,liq_cross_duration=2.57:/iwradio/arquivos/vinhetas/64.mp3**
> 2024/07/08 17:01:42 [lang:2] ---- Adding: annotate:liq_cue_in=0.,liq_cross_duration=1.62:/iwradio/arquivos/musicas/2407/11457.mp3
> 2024/07/08 17:01:52 [decoder:2] Decoding "/iwradio/arquivos/musicas/2407/11456.mp3" ended: Mad.End_of_stream.
> 2024/07/08 17:01:52 [cross:1] Invalid cross duration: 0.00 <= 0.!
> 2024/07/08 17:01:52 [decoder:2] Decoding "/iwradio/arquivos/vinhetas/64.mp3" ended: Mad.End_of_stream.
> **2024/07/08 17:01:52 [cross:1] End of track reached while buffering next track data, crossfade duration is longer than the track's duration. Make sure to adjust the crossfade duration to avoid issues.**

But "64.mp3" is >5 seconds. The trigger point is 2.57s before the end of the station ID with >5s.
